src/main/java/usi/si/seart/staticcodeanalysis/TerminalExecution.java

```diff
@@ -0,0 +1,157 @@
+
+        switch (System.getProperty("os.name")) {
+            case "windows":
+                processBuilder.command("cmd.exe", "/c", String.join(" ", args));
+                break;
+            default:
+                processBuilder.command("bash", "-c", String.join(" ", args));
+                break;
+        }
+    }
+
+
+    /**
+     * Invokes the command in the shell.
+     *  If a process was already invoked and is still running, it first stops it.
+     * @throws TerminalExecutionException Thrown if an error occurred when invoking the command.
+     */
+    public TerminalExecution start() throws TerminalExecutionException {
+        if (isRunning())
+            stop();
+        try {
+            process = processBuilder.start();
+        } catch (IOException e) {
+            throw new TerminalExecutionException("Could not start terminal process",e);
+        }
+        return this;
+    }
+
+    /**
+     * Terminates the process.
+     *  If the process is not running, does nothing.
+     */
+    public TerminalExecution stop() {
+        if(process == null)
+            return this;
+        process.destroy();
+        return this;
+    }
+
+    /**
+     * Returns a BufferedReader for reading from the process' normal output.
+     * @return the BufferedReader for reading the normal output piped from the process.
+     */
+    public BufferedReader getStdout() {
+        return new BufferedReader(new InputStreamReader(process.getInputStream()));
+    }
+
+
+    public BufferedReader getStdErr() {
+        return new BufferedReader(new InputStreamReader(process.getErrorStream()));
+    }
+
+    public boolean isRunning() {
+        return process != null && process.isAlive();
+    }
+
+    /**
+     * Waits for the process to end.
+     *  Method returns without error regardless of the process terminating abnormally, or if the waiting thread is interrupted.
+     */
+    public TerminalExecution waitEnd() {
+        if (process == null)
+            return this;
+
+        try {
+            process.onExit().get();
+        } catch (ExecutionException | CancellationException e) {
+
+        } finally {
+            return this;
+        }
+    }
+
+    /**
+     * Waits for the successful execution of the process.
+     * @throws RuntimeException If the command exits with an error or throws any kind of exception.
+     */
+    public TerminalExecution waitSuccessfulExit() throws TerminalExecutionException {
+        if (process == null)
+            return this;
+
+        int exitCode = 0;
+        BufferedReader stderr;
+        try {
+            stderr = new BufferedReader(new InputStreamReader(process.getErrorStream()));
+            exitCode = process.onExit().get().exitValue();
+            if (exitCode != 0) {
+                throw new Exception("Terminal process returned error code "+exitCode+"\n== stderr:\n"+stderr.lines().collect(Collectors.joining("\n"))+"\n===");
```

Comment:
Separate this into multiple statements: one where you join the lines, one where you format the string, and one where you throw the actual exception. This way it will be easier to comprehend